### ISSUE TITLE
[Feature Request]: Ability to enable Ice Bar only when on built-in display

### ISSUE THREAD

**Issue by kemayo** [jordanbaird/Ice#223]
### CHECK EXISTING ISSUES

- [X] I have searched existing issues, and this feature has not been requested

### Description

The new Ice Bar feature in 0.10 is a great way to avoid icons getting hidden under the notch. However, I only care about that when I'm using my laptop's built-in display -- when I'm using an external monitor I will almost-never have to worry about icons colliding with the application menus.

It would be nice if the Ice Bar setting could be made conditional -- "Use Ice Bar: Yes / No / Only on built-in display". (Or "only when there's a notch", but the built-in displays are generally smaller anyway, so...)

This is similar to #188, in that it's a "depending on whether we're on a screen with a notch" behavior-change. It's possible that a more-comprehensive solution would be to duplicate a large chunk of the behavior / appearance settings into two separate "built-in" / "external" sections.

**Comment by ArlinJae** [comment]
Seconded this — think this would be really cool!

**Comment by bjorntj** [comment]
Yes, this is the only thing that is missing. I now stop the application when using my external display but I have icons I would like to have always hidden, so would ne nice with a feature to show all the icons, except for always hidden, when display resolution are larger than x, or something like that.

**Comment by claytonolley** [comment]
After moving to this app from Bartender, this is the biggest feature that seems to be missing... when I'm on my ultrawide monitors, I don't need to hide this stuff.

**Comment by sobbybutter** [comment]
This is the only missing feature for me. Would love to see it implemented!

**Comment by mgrn0** [comment]
second all the comments above, this would be SO useful!

**Comment by fosple** [comment]
This is definitely the feature missing compared to Bartender.

**Comment by manuch1176** [comment]
+1 on this, it's the only missing thing that keeps me moving from Bartender (and then paying for the app).

**Comment by fredngo** [comment]
Already migrated from (and deleted) Bartender before realizing this was missing.

Still gonna use Ice, but would very much like to see this happen!

**Comment by matthewdias** [comment]
Can future folks that want to show support for this request please 👍🏽 thumbs up the [original post](https://github.com/jordanbaird/Ice/issues/223#issue-2398318714) instead of leaving a +1 comment? That way we can all stay subscribed for when notable updates happen without additional noise. Thanks!

**Comment by pourmand1376** [comment]
This is very needed. On M2 Air Macbooks, we have a lot of black area around the webcam, so I can not even see ice. 

<img width="1436" height="156" alt="Image" src="https://github.com/user-attachments/assets/6ba6cac5-1bd4-46c1-9ce2-80a82a00f722" />

**Comment by pourmand1376** [comment]
Today, I took some time and actually implemented this! 

The DMG file is ready to be used (available in the PR). It is not fully tested or anything. But works for me, Let me know if you have any ideas.

**Comment by pourmand1376** [comment]
> I tried to install it, however Ice kept asking me for (screen recording and accessibility) and I could not get beyond that point, no matter what (of course, granted these but no change). putting the last beta back in place and everything worked again..

I also had this problem. You have to first delete all permissions of ice and also delete its row (via the minus button). 

- Privacy and Security -> Accessibility
- Privacy and Security -> Screen and Audio Recording

<img width="919" height="1093" alt="Image" src="https://github.com/user-attachments/assets/79976b2b-b8b5-4aa4-8759-ffddff63ffaf" />

<img width="919" height="1093" alt="Image" src="https://github.com/user-attachments/assets/c30231d9-82b5-4720-b89a-843f45b30df3" />

The reason is these permissions are granted for main ice app. But my build is not signed or anything. So, we should delete all permissions and give it again.

**Comment by mgrn0** [comment]
Any progress on this? Would be so nice to have this in the regular release … the build file is not available anymore, unfortunately.

**Comment by benjuade** [comment]
+1 on this. I've found other issues opened that can be merged to this one:

- https://github.com/jordanbaird/Ice/issues/797
- https://github.com/jordanbaird/Ice/issues/703
- https://github.com/jordanbaird/Ice/pull/795
- https://github.com/jordanbaird/Ice/pull/612

**Comment by cbruegg** [comment]
Thaw, a fork of Ice, seems to have this feature: https://github.com/stonerl/Thaw

**Comment by Caffeine19** [comment]
> Thaw, a fork of Ice, seems to have this feature: https://github.com/stonerl/Thaw

Thaw is great, but have more bugs on macOS15. I still wish Ice could add this feat. 😭
